### ISSUE TITLE
Building for Solaris

### ISSUE THREAD

**Issue by UnixRonin** [unixwork/xnedit#73]
The README says that:

[...] the requirements to build XNEdit from the
sources are:

 - ANSI C99 compiler
 - make utility (eg, GNU make)
 - X11 development stuff (headers, libraries)
 - Xrender and Xft
 - Fontconfig
 - Motif 2.0 or above

Optionally one may use:

 - yacc (or GNU bison)

It then says that the command to build on Solaris is:

make solaris

On a new Solaris 11.4 install with Oracle Solaris Developer Suite 12.6, this yields the following:

epsilon3:root:/mnt/morpheus/export/src/xnedit:13 # make solaris
(cd util; \
    make -f Makefile.solaris libNUtil.a)
sh: pkg-config: not found
mksh: Fatal error: The command `pkg-config --cflags xft fontconfig' returned status `127'
Current working directory: /mnt/morpheus/export/src/xnedit/util
*** Error code 1
make: Fatal error: Command failed for target `solaris'

Apparently there is an assumption here that pkg-config will be present which is not actually present on Solaris 11.4.
Is it assumed that this will be installed by whatever source "X11 development stuff (headers, libraries) [including] Xrender and Xft" is installed from?

(I'm actually totally good with using a pre-built Solaris binary, if I could find one.)

**Comment by unixwork** [comment]
You are right, pkg-config should be listed as a dependency. It is such an elementary and important tool, that I sometimes forget, it needes to be installed.

On Solaris 11, you can install it with `pkg install pkg-config`.

Binary packages are on my todo list, but I think it will take some time.

**Comment by UnixRonin** [comment]
Got that, thanks.
I also found, after fixing that, that where the requirements say "ANSI C99 compiler", what is really meant is gcc.  It will not even configure using Solaris CC.  Solaris 11.4 does however have a gcc package available, though it appears it's only gcc 7.3.  '`pkg install gcc`' will install gcc and binutils.

So far I have the following packages installed:
solaris-desktop
gcc
pkg-config
motif

And that gets me this far:
```
gcc -c -I../Microline -I../Xlt -O -std=gnu99 -m64 -I/usr/openwin/include -I/usr/include/X11 -I/usr/include/pcre -DUSE_DIRENT -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -I/usr/include/freetype2 -I/usr/include/libpng14 -o window.o window.c
window.c:112:10: fatal error: Xm/DropDown.h: No such file or directory
 #include <Xm/DropDown.h>
          ^~~~~~~~~~~~~~~
```

Have you any idea how to resolve this?

**Comment by unixwork** [comment]
I should have tested building on Solaris before making a release...

The newest commit should fix the problem.

Also, it is possible to build with suncc, however it would require some manual adjustments in the file `makefiles/Makefile.solaris`. You need to change the CC variable and remove `-std=gnu99 ` from the CFLAGS variable.

Or just build with gcc. I though this would be a reasonable default, because it also work on IllumOS distributions.

**Comment by unixwork** [comment]
I've noticed the Color preference dialog is unusable on Solaris 10. Not sure if Solaris 11 is also affected, I will test that later. The problem only occures in xnedit 1.4, xnedit 1.3.3 should work fine.

**Comment by UnixRonin** [comment]
> I should have tested building on Solaris before making a release...
> 
> The newest commit should fix the problem.

And does indeed.  Thanks!  Even the mouse pointer is a usable size on a 4K display right out of the box.

> Also, it is possible to build with suncc, however it would require some manual adjustments in the file `makefiles/Makefile.solaris`. You need to change the CC variable and remove `-std=gnu99 ` from the CFLAGS variable.
> 
> Or just build with gcc. I though this would be a reasonable default, because it also work on IllumOS distributions.

Yup, it works fine with the gcc package available on the Solaris 11.4 repo, given the above fix.
